UDP crash

```
2015/02/15 14:57:13 [TXP] Known transaction (28bd85bf)
2015/02/15 14:57:13 [TXP] Known transaction (de4ea2b6)
2015/02/15 14:57:13 [TXP] Known transaction (eac76bc8)
2015/02/15 14:57:13 [TXP] Known transaction (a94416db)
2015/02/15 14:58:14 [Peer ec38655e49a561b3 208.52.168.126:49424] whisper stopped
panic: runtime error: slice bounds out of range

goroutine 24 [running]:
github.com/ethereum/go-ethereum/p2p/discover.(*udp).loop(0xc2080d0b40)
	/Users/jeffrey/go/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:259 +0x852
created by github.com/ethereum/go-ethereum/p2p/discover.ListenUDP
	/Users/jeffrey/go/src/github.com/ethereum/go-ethereum/p2p/discover/udp.go:155 +0x5a6
```

--- a/p2p/discover/udp.go
+++ b/p2p/discover/udp.go
@@ -253,7 +253,8 @@ func (t *udp) loop() {
 
 		case reply := <-t.replies:
 			// run matching callbacks, remove if they return false.
-			for i, p := range pending {
+			for i := 0; i < len(pending); i++ {
+				p := pending[i]
 				if reply.from == p.from && reply.ptype == p.ptype && p.callback(reply.data) {
 					p.errc <- nil
 					copy(pending[i:], pending[i+1:])

